rpc-server/src/dispatchers/blockchain.rs

```diff
@@ -529,4 +588,152 @@ impl BlockchainInterface for BlockchainDispatcher {
+
+        Ok(stream
+            .filter_map(move |event| {
+                let result = match event {
+                    BlockchainEvent::EpochFinalized(hash) => {
+                        let blockchain_rg = blockchain.read();
+                        Some(
+                            get_validator_by_address(&blockchain_rg, &address, Some(false))
+                                .map_err(|_| hash),
+                        )
+                    }
+                    _ => None,
+                };
+                future::ready(result)
+            })
+            .boxed())
+    }
+
+    /// Subscribes to log events.
+    #[stream]
+    async fn logs_subscribe(&mut self) -> Result<BoxStream<'static, BlockLog>, Self::Error> {
+        let stream = self.blockchain.write().log_notifier.as_stream();
+
+        Ok(stream.boxed())
+    }
+
+    /// Subscribes to log events related to a given list of addresses.
+    #[stream]
+    async fn logs_by_addresses_subscribe(
+        &mut self,
+        addresses: Vec<Address>,
+    ) -> Result<BoxStream<'static, BlockLog>, Self::Error> {
+        let stream = self.logs_subscribe().await?;
+
+        Ok(stream
+            .filter_map(move |event| {
+                let result = match event {
+                    BlockLog::AppliedBlockLog {
+                        mut inherent_logs,
+                        block_hash,
+                        block_number,
+                        mut tx_logs,
+                    } => {
+                        let mut block_log = None;
+                        inherent_logs.retain(|log| is_log_related_to_addresses(&log, &addresses));
+                        tx_logs.retain(|tx_log| {
+                            is_tx_logs_related_to_any_addresses(&tx_log, &addresses)
+                        });
+                        if !inherent_logs.is_empty() || !tx_logs.is_empty() {
+                            block_log = Some(BlockLog::AppliedBlockLog {
+                                inherent_logs,
+                                block_hash,
+                                block_number,
+                                tx_logs,
+                            });
+                        }
+                        block_log
+                    }
+                    BlockLog::RevertBlockLog {
+                        mut inherent_logs,
+                        block_hash,
+                        block_number,
+                        mut tx_logs,
+                    } => {
+                        let mut block_log = None;
+                        inherent_logs.retain(|log| is_log_related_to_addresses(&log, &addresses));
+                        tx_logs.retain(|tx_log| {
+                            is_tx_logs_related_to_any_addresses(&tx_log, &addresses)
+                        });
+                        if !inherent_logs.is_empty() || !tx_logs.is_empty() {
+                            block_log = Some(BlockLog::AppliedBlockLog {
```

Comment:
This is in the `BlockLog::RevertBlockLog` matching block, so I think:
```suggestion
                            block_log = Some(BlockLog::RevertBlockLog {
```

Reply:
You are 100% correct. Thank you!